### DISCUSSION TITLE
Plans for date field / date picker component?

### DISCUSSION THREAD

**Discussion by jgreenblatt-JPM** [jpmorganchase/salt-ds#3050] [Feedback]
Hi team, 

Part of the experience I'm designing requires a user to enter their date of birth. In the short term, with the aim of sticking with salt DS components as much as possible, I've suggested 3, grouped input fields (day, month, year).

That said, it would be more ideal to have a single input field for capturing a date and also the date picker UI as secondary option. For example in MUI: https://mui.com/x/react-date-pickers/date-picker/

Are there any plans to add similar functionality to Salt?

Thanks,
Jonathan

**Comment by origami-z** [comment]
Date picker is in the work (#2878, #2877, #2881).

IMO, DOB is better to be captured separately (maybe dropdown) than date pickers.

**Reply by jgreenblatt-JPM** [reply]
IMO, DOB as a single line date formatted input with date picker as a secondary options would be the ideal. 3 drop drop downs seems like a lot of interaction. Designing UX: Forms by Jessica Enders is a good resource on this topic. Would love to understand your opinion further. Maybe we can catchup offline about it.

**Reply by origami-z** [reply]
Everyone has different opinion on this. For me, date picker thrives for controlled range / short term / close & recent dates / pre-selected dates and date ranges, e.g. holiday / flights. DOB is quite the opposite, format / locale input order, number v.s. letter for month, keyboard vs mobile, auto complete complexity, etc. 

There's a reason gov.uk chose to use 3 separate inputs instead of 1 input and/or fancier pickers.

**Reply by jgreenblatt-JPM** [reply]
Nice! Thanks for providing your thoughts. This blog post from UK gov design system team explains their evolution on DOB input--they started with 3 dropdowns and gradually moved to 3 inputs. 

https://designnotes.blog.gov.uk/2013/12/05/asking-for-a-date-of-birth/

**Comment by mark-tate** [comment]
Thanks for the creating the discussion the initial Date Controls are now in PR

https://github.com/jpmorganchase/salt-ds/pull/3234

Closing this discussion and directing any further feedback to the PR/issue
